### PR TITLE
Fix: support only provide vela or kubernetes version requirement and generate this annotation when package addon

### PR DESCRIPTION
**Pull request by wangyikewxgm**
Signed-off-by: 楚岳 <wangyike.wyk@alibaba-inc.com>

1. Addons only provide only one of vela or kubernetes system requirement.
2. `vela addon package` command generate system requirement annotation.

```yaml
apiVersion: v1
entries:
    dex:
        - annotations:
              system.vela: ">=v1.3.0-beta.1"
          apiVersion: v2
          appVersion: 0.6.5
          created: "2022-06-30T09:07:10.5877978Z"
          description: Enable dex for login
          digest: c6a870a9a17c95dcaa15a5803f9c8b8fa852b760a654c109451e3ff1677a7b28
          icon: https://raw.githubusercontent.com/cncf/artwork/master/projects/dex/horizontal/color/dex-horizontal-color.png
          keywords:
              - auth
              - alpha
          name: dex
          type: library
          urls:
              - charts/dex-0.6.5.tgz
          version: 0.6.5
generated: "2022-06-30T09:07:11Z"
serverInfo: {}
```

### Description of your changes

Fixes #

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/addon/addon.go [hunk 1]
```diff
@@ -1393,7 +1393,7 @@ func (h *Installer) getAddonVersionMeetSystemRequirement(addonName string) strin
 			return ""
 		}
 		for _, version := range versions {
-			req := LoadSystemRequirements(version.Annotations["system"])
+			req := LoadSystemRequirements(version.Annotations)
```
Comment (barnettZQG) [inline pkg/addon/addon.go:1396]: Why change the spec?

Comment (wangyikewxgm) [reply pkg/addon/addon.go:1396]: Addon maybe only have one type of system requirement(vela or kubernetes),should not store them in one annotation.

**Review by wonderflow — APPROVED** [review]
good job

File: e2e/addon/mock/testrepo/helm-repo/index.yaml [hunk 2]
```diff
@@ -10,7 +10,8 @@ entries:
         - http://127.0.0.1:9098/helm/fluxcd-test-version-1.0.0.tgz
       version: 1.0.0
       annotations:
-        system: "vela>=1.3.0; kubernetes>=1.10.0"
+        system.vela: ">=1.3.0"
```
Comment (StevenLeiZhang) [inline e2e/addon/mock/testrepo/helm-repo/index.yaml:13]: This is mocked Chart repo and index file. Do the two fields work for official Chartmesum or Harbor

Comment (wangyikewxgm) [reply e2e/addon/mock/testrepo/helm-repo/index.yaml:13]: Yes, have tested. Cause `annotation` can be set with any key-value pair

File: pkg/addon/addon.go (hunk 1 above)
Comment (barnettZQG) [reply pkg/addon/addon.go:1396]: Are there backward compatibility issues?

Comment (wangyikewxgm) [reply pkg/addon/addon.go:1396]: No, the CI of catalog have not add this mechanism. Refer to: https://github.com/kubevela/catalog/pull/397

**Review by Somefive — APPROVED** [review]

**Pull request merged**